rsa_utils.go

```diff
@@ -91,7 +92,9 @@ func ParseRSAPublicKeyFromPEM(key []byte) (*rsa.PublicKey, error) {
 		if cert, err := x509.ParseCertificate(block.Bytes); err == nil {
 			parsedKey = cert.PublicKey
 		} else {
-			return nil, err
+			if parsedKey, err = x509.ParsePKCS1PublicKey(block.Bytes); err != nil {
+				return nil, fmt.Errorf("%v - %w", ErrNotRSAPublicKey.Error(), err)
```

Comment:
this sort of breaks our existing error pattern. this should probably just return `ErrKeyMustBePEMEncoded`.